### DISCUSSION TITLE
Subset has imbalanced number of equations (really large model)

### DISCUSSION THREAD

**Discussion by bflo96** [OpenModelica/OpenModelica#12846] [General]
Hi everyone,

I was running my model and, despite the number of equations and variables being the same when I check the model, it gives:

`An independent subset of the model has imbalanced number of equations (34618) and variables (34682).`

What is the most adequate procedure when this happens to check what might have gone wrong considering the large number of variables and equations I am dealing with?

I can give more details if needed, but first I was wondering if there is any kind of checklist I could go through of usual tips or errors.

Thank you in advance, your help is much appreciated.

**Comment by bflo96** [comment]
Closing this as I managed to find the problem in my model.

**Comment by casella** [comment]
In general, Modelica 3.0 introduced restriction on balanced connectors, that allow to clearly identify balanced components. It can then be proven that the connection of balanced models provides balanced system models. So, you can check the individual components (and sub-components) of your model one by one. Divide et impera.

In fact, we could improve the checking so that this is done automatically. That's in our long-term todo list.
